addons/portal/models/res_partner.py

```diff
@@ -1,12 +1,34 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import fields, models
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
+    is_default_billing_address = fields.Boolean(default=False, store=True)
+    is_default_shipping_address = fields.Boolean(default=False, store=True)
+
+    def _update_delivery_and_shipping_address(self, partner_id, address_type, **kw):
+        partner_sudo = self.browse(partner_id)
+
+        address_type = 'invoice' if address_type == 'invoice' or address_type == 'billing' else 'delivery'
+        default_field = 'is_default_billing_address' if address_type == 'invoice' else 'is_default_shipping_address'
+        if partner_sudo.parent_id:
+            partner_sudo.parent_id.child_ids.filtered(lambda p: p.type == address_type or p.type == 'other').write({default_field: False})
+            partner_sudo.parent_id.child_ids.filtered(lambda p: p.type == address_type).write({'type': 'other'})
```

Comment:
Could just get child_ids here and only have the filtering once after the if blocks, would avoid having to duplicate filters